### PR TITLE
[jsk_perception] Enable ApplyMask handle BGRA/RGBA image

### PR DESCRIPTION
**Pull request by wkentaro**
Now ApplyMask always publish RGB8 image

### CODE REVIEW

**Comment by garaemon** [comment]
We should keep original format to support RGB, BGR and mono

**Comment by wkentaro** [comment]
I see.
How about RGBA or BGRA?
Can I convert it to RGB or BGR?

**Comment by garaemon** [comment]
You mean you need to handle the image which has alpha channel?

**Comment by wkentaro** [comment]
Input image's encoding is rgba8 so applying mono8 mask to rgba8 does not work.
I'd like apply_mask to support rgba channel image, by converting it to rgb image.

In my case, Baxter's right hand camera image is rgb8..
image

**Comment by wkentaro** [comment]
I updated to keep input image encoding.

File: jsk_perception/src/apply_mask_image.cpp [hunk 1]
```diff
@@ -77,8 +78,25 @@ namespace jsk_perception
...
+      if (image_msg->encoding == sensor_msgs::image_encodings::BGRA8 ||
+          image_msg->encoding == sensor_msgs::image_encodings::BGRA16) {
+        cv::cvtColor(image_raw, image, cv::COLOR_BGRA2BGR);
+        image_raw.copyTo(image);
+      } else if (image_msg->encoding == sensor_msgs::image_encodings::RGBA8 ||
+                 image_msg->encoding == sensor_msgs::image_encodings::RGBA16) {
+        cv::cvtColor(image_raw, image, cv::COLOR_RGBA2BGR);
+        image_raw.copyTo(image);
+      }
+    } else {
```
Comment (wkentaro) [inline jsk_perception/src/apply_mask_image.cpp]: This if{} block is to convert RGBA -> RGB, BGRA->BGR.

**Comment by garaemon** [comment]
Hmm, it's difficult.
If we can write more generinc node to run `cv::cvtColor`, it would be the best.
anyway, I merged it

**Pull request merged**